tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -224,62 +227,42 @@ public async Task Sanitize(IEnumerable<RecordedTestSanitizer> sanitizers, bool s
+        { 
+            var logMessage = (sanitizerId != null && sanitizerId.StartsWith("AZSDK") ? $"Central sanitizer " : "User specified ") + $"rule {sanitizerId} modified the entry{Environment.NewLine}";
+            var before = $"{Environment.NewLine}before:{Environment.NewLine} ";
+            var after = $"{Environment.NewLine}after: {Environment.NewLine} ";
+            if (entryPostSanitize.RequestUriIsModified)
             {
-                DebugLogger.LogDebug($"RequestUri is modified{Environment.NewLine}before:{Environment.NewLine} {entryPreSanitize.RequestUri}{Environment.NewLine}after:{Environment.NewLine} {entryPostSanitize.RequestUri}{Environment.NewLine}");
+                logMessage+= $"RequestUri is modified{before}{entryPreSanitize.RequestUri}{after}{entryPostSanitize.RequestUri}{Environment.NewLine}";
             }
 
             if (entryPostSanitize.Request.IsModified.Headers)
             {
-                DebugLogger.LogDebug($"Request Headers are modified{Environment.NewLine}before:{Environment.NewLine} {entryPreSanitize.Request.Headers}{Environment.NewLine}after:{Environment.NewLine} {entryPostSanitize.Request.Headers}{Environment.NewLine}");
+                logMessage += $"Request Headers are modified{before}{entryPreSanitize.Request.Headers.ToString()}{after}{entryPostSanitize.Request.Headers.ToString()}{Environment.NewLine}";
             }
 
             if (entryPostSanitize.Response.IsModified.Headers)
             {
-                DebugLogger.LogDebug($"Response Headers are modified{Environment.NewLine}before:{Environment.NewLine} {entryPreSanitize.Response.Headers}{Environment.NewLine}after:{Environment.NewLine} {entryPostSanitize.Response.Headers}{Environment.NewLine}");
+                logMessage += $"Response Headers are modified{before}{entryPreSanitize.Response.Headers.ToString()}{after}{entryPostSanitize.Response.Headers.ToString()}{Environment.NewLine}";
             }
 
             if (entryPostSanitize.Request.IsModified.Body)
             {
-                DebugLogger.LogDebug($"Request Body is modified{Environment.NewLine}before:{Environment.NewLine} {entryPreSanitize.Request.Body}{Environment.NewLine}after:{Environment.NewLine} {entryPostSanitize.Request.Body}{Environment.NewLine}");
+                logMessage += $"Request Body is modified{before}{entryPreSanitize.Request.Body.ToString()}{after}{entryPostSanitize.Request.Body.ToString()}{Environment.NewLine}";
             }
 
             if (entryPostSanitize.Response.IsModified.Body)
             {
-                DebugLogger.LogDebug($"Response Body is modified{Environment.NewLine}before:{Environment.NewLine} {entryPreSanitize.Response.Body}{Environment.NewLine}after:{Environment.NewLine} {entryPostSanitize.Response.Body}{Environment.NewLine}");
+                logMessage += $"Response Body is modified{before}{entryPreSanitize.Response.Body.ToString()}{after}{entryPostSanitize.Response.Body.ToString()}{Environment.NewLine}";
```

Comment:
TODO: use something other than `.ToString()` for detailed logging of before and after.

Reply:
It's dangerous to go alone, [take this](https://github.com/Azure/azure-sdk-tools/blob/3e14dcaf28df0f1c81948114c2016b28a9991c83/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RequestOrResponse.cs#L39)
```
if (entryPostSanitize.Response.TryGetBodyAsText(out string bodyText)) {
   <use the string>
}.
```

Reply:
updated with this